src/main/java/com/puppycrawl/tools/checkstyle/checks/SuppressWarningsHolder.java

```diff
@@ -398,9 +399,11 @@ private static String getStringExpr(DetailAST ast) {
                     return quotedText.substring(1, quotedText.length() - 1);
                 case TokenTypes.IDENT:
                     return firstChild.getText();
+                case TokenTypes.DOT:
+                    return firstChild.getLastChild().getText();
                 default:
-                    throw new IllegalArgumentException("String literal AST expected: "
-                            + firstChild);
+                    // annotations with complex expressions cannot suppress warnings
+                    return "";
```

Comment:
This is controversial, but what else can we do if user suppresses warnings using something like `@SuppressWarnings((1 != 1) ? "" : "unused")`?